### ISSUE TITLE
RECL not updated everytime

### ISSUE THREAD

**Issue by harshg99** [NJU-RL/ACORM#6]
in the update_recl function where the CL loss is computed, why are we training the role embedding network only if batch_active[idx, t] > 0.5 ?

**Comment by huzican** [comment]
Because batch_active[idx, t] is a binary value (either 0 or 1). It is used to determine whether the trajectory at that specific time step is valid or not.

**Comment by harshg99** [comment]
yes, I agree with that, do we need to check its validity when doing the loss backdrop?

**Comment by huzican** [comment]
Initially, the goal was to reduce the number of backpropagation steps. However, we found that the difference in runtime was actually not significant.
